ban-cooking.lua

```diff
@@ -0,0 +1,440 @@
+    for _, p in ipairs(df.global.world.raws.plants.all) do
+        for _, m in ipairs(p.material) do
+            if m.flags.ALCOHOL and m.flags.EDIBLE_COOKED then
+                local matinfo = dfhack.matinfo.find(p.id, m.id)
+                ban_cooking(p.name .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.DRINK, -1)
+            end
+        end
+    end
+    for _, c in ipairs(df.global.world.raws.creatures.all) do
+        for _, m in ipairs(c.material) do
+            if m.flags.ALCOHOL and m.flags.EDIBLE_COOKED then
+                local matinfo = dfhack.matinfo.find(creature_id.id, m.id)
+                ban_cooking(c.name[2] .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.DRINK, -1)
+            end
+        end
+    end
+end
+
+funcs["honey"] = function()
+    local mat = dfhack.matinfo.find("CREATURE:HONEY_BEE:HONEY")
+    ban_cooking('honey bee honey', mat.type, mat.index, df.item_type.LIQUID_MISC, -1)
+end
+
+funcs["tallow"] = function()
+    for _, c in ipairs(df.global.world.raws.creatures.all) do
+        for _, m in ipairs(c.material) do
+            if m.flags.EDIBLE_COOKED then
+                for _, s in ipairs(m.reaction_product.id) do
+                    if s.value == "SOAP_MAT" then
+                        local matinfo = dfhack.matinfo.find(c.creature_id, m.id)
+                        ban_cooking(c.name[2] .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.GLOB, -1)
+                        break
+                    end
+                end
+            end
+        end
+    end
+end
+
+funcs["milk"] = function()
+    for _, c in ipairs(df.global.world.raws.creatures.all) do
+        for _, m in ipairs(c.material) do
+            if m.flags.EDIBLE_COOKED then
+                for _, s in ipairs(m.reaction_product.id) do
+                    if s.value == "CHEESE_MAT" then
+                        local matinfo = dfhack.matinfo.find(c.creature_id, m.id)
+                        ban_cooking(c.name[2] .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.LIQUID_MISC, -1)
+                        break
+                    end
+                end
+            end
+        end
+    end
+end
+
+funcs["oil"] = function()
+    for _, p in ipairs(df.global.world.raws.plants.all) do
+        for _, m in ipairs(p.material) do
+            if m.flags.EDIBLE_COOKED then
+                for _, s in ipairs(m.reaction_product.id) do
+                    if s.value == "SOAP_MAT" then
+                        local matinfo = dfhack.matinfo.find(p.id, m.id)
+                        ban_cooking(p.name .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.LIQUID_MISC, -1)
+                        break
+                    end
+                end
+            end
+        end
+    end
+end
+
+funcs["seeds"] = function()
+    for _, p in ipairs(df.global.world.raws.plants.all) do
+        if p.material_defs.type.seed ~= -1 and p.material_defs.idx.seed ~= -1 and not p.flags.TREE then
```

Comment:
you can invert the condition and use `goto continue` for this one to remove an entire level of indentation from the function